src/lib.rs

```diff
@@ -134,48 +147,59 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		#[pallet::weight(1)]
-		pub fn create_subsciption_plan(
-			_origin: OriginFor<T>,
-			_plan: Subscription<T::BlockNumber, BalanceOf<T>, T::AccountId>,
+		pub fn create_subscription_plan(
+			origin: OriginFor<T>,
+			plan: SubscriptionPart<T::BlockNumber, BalanceOf<T>, T::AccountId>,
 		) -> DispatchResult {
-			// maybe make a new function for Subscription, so `plan` is already valid 100%
-			//
-			//
-			// todo:
-			// signed ?
-			// valid plan ?
-			//
-			// -> push the plan to the storage
+			let who = ensure_signed(origin)?;
+
+			if plan.frequency == 0u32.into() {
+				return Err(Error::<T>::BadFrequency.into())
+			}
+			if plan.beneficiary != who {
+				return Err(Error::<T>::Unauthorized.into())
+			}
+
+			let nonce = Self::plan_nonce();
+
+			PlanNonce::<T>::put(nonce.saturating_add(1));
```

Comment:
`checked_add` rather than saturating.
We don't want two plans to end up having the same Id. We want to stop adding new ones if we don't have available ids